src/stan/lang/grammars/statement_grammar_def.hpp

```diff
@@ -327,15 +337,15 @@ namespace stan {
         > -expression_g(_r1)
         > lit(']');
 
-      // _r1 = allow sampling, _r2 = var origin
+      // inherited feature _r1 = var_origin
       return_statement_r.name("return statement");
       return_statement_r
         %= (lit("return") >> no_skip[!char_("a-zA-Z0-9_")])
         >> expression_g(_r1)
         >> lit(';') [validate_return_allowed_f(_r1, _pass,
                                        boost::phoenix::ref(error_msgs_))];
 
-      // _r1 = var origin
+      // inherited feature _r1 = var_origin
```

Comment:
Yes, please leave all these comments---they're super helpful when scanning the rules.  Just drop the "inherited feature" part of the comment because `_r1` is an inherited feature by definition.